src/API/Reports/DataStore.php

```diff
@@ -66,6 +66,109 @@ class DataStore {
 	 * @var string
 	 */
 	private $order = '';
+	/**
+	 * Data store context used to pass to filters.
+	 *
+	 * @var string
+	 */
+	private $context = 'reports';
+	/**
+	 * List of SQL clauses.
+	 *
+	 * @var array
+	 */
+	private $sql_clauses = array(
+		'select'     => array(),
+		'from'       => array(),
+		'outer_from' => array(),
+		'where'      => array(),
+		'where_time' => array(),
+		'limit'      => array(),
+		'order_by'   => array(),
+	);
+
+	/**
+	 * Assign table name in database class
+	 */
+	protected function set_db_table_name() {
+		global $wpdb;
+		$class = get_called_class();
+		if ( $class::TABLE_NAME ) {
+			$wpdb->{$class::TABLE_NAME} = $wpdb->prefix . $class::TABLE_NAME;
+		}
+	}
+
+	/**
+	 * Get table name from database class
+	 */
+	protected function get_db_table_name() {
+		global $wpdb;
+		$class = get_called_class();
+		return isset( $wpdb->{$class::TABLE_NAME} ) ? $wpdb->{$class::TABLE_NAME} : $class::TABLE_NAME;
+	}
+
+	/**
+	 * Prepend a query field name with the data store's table name.
+	 *
+	 * @param string $query      Query string to search.
+	 * @param string $field_name The target field name.
+	 *
+	 * @return string Updated query string.
+	 */
+	protected function prepend_table_name( $query, $field_name ) {
+		return str_replace( $field_name, $this->get_db_table_name() . '.' . $field_name, $query );
+	}
+
+	/**
+	 * Add a SQL clause to be included when get_data is called.
+	 *
+	 * @param string $clause SQL clause.
+	 * @param string $type   (select|from|outer_from|where|where_time|order_by|limit).
+	 */
+	protected function add_sql_clause( $clause, $type ) {
```

Comment:
What do you think about switching the argument order here?

`$this->add_sql_clause( 'where', '1=1' );` reads a bit more natural to me than `$this->add_sql_clause( '1=1', 'where' );`

Reply:
> What do you think about switching the argument order here?

Great idea. Much more readable.

Reply:
Parameters swapped in https://github.com/woocommerce/woocommerce-admin/pull/2961/commits/9e260aa6e9fcdb990219a70287d4db4c9179f9a8

Reply:
parameters swapped in https://github.com/woocommerce/woocommerce-admin/pull/2961/commits/9e260aa6e9fcdb990219a70287d4db4c9179f9a8